src/lib/api.ts

```diff
@@ -107,6 +119,29 @@ export class AeriesApi {
 				}/homescreendata`
 			)
 		);
+		if (!resp.ok) {
+			let as = get(this.authedStudent)!; // should never be null since isInitialized() is true;
+			let user = as.Username;
+			let pass = as.Password;
+			this.authedStudent.set(null);
+			let authed = await this.authenticate(as.Username, as.Password);
+			console.log(
+				`Attempted re-authentication due to token expiration. Res=${authed}`
+			);
+			if (authed) {
+				// just re-try it
+				resp = await fetch(
+					this.genRequest(
+						"GET",
+						`/student/${
+							get(this.authedStudent)!.Student.Demographics.StudentID
+						}/homescreendata`
+					)
+				);
```

Comment:
This works, ig we could also return the user to login again, but honestly I find this annoying, so I think this is fine.